client/src/partials/fiscal/fiscal.create.js

```diff
@@ -1,427 +1,446 @@
+    if ($scope.data.start < $scope.data.end) {
       $scope.step = steps[hasPreviousYear ?  1 : 2];
       if (hasPreviousYear) {
         session.previous_fiscal_year = $scope.fiscal.get(data.previous_fiscal_year).fiscal_year_txt;
       }
+    } else {
+      var msg = $translate.instant('FISCAL_YEAR.DATE_ORDER_ERROR')
+      .replace('%START%', util.htmlDate($scope.data.start))
+      .replace('%END%', util.htmlDate($scope.data.end));
+      messenger.error(msg);
     }
 
-    function checkClosing (close_fy, previous_fy_id) {
-      if (close_fy) {
-        stepThree(previous_fy_id);
-      } else {
-        $scope.createWithoutClosing = true;
-        $scope.step = steps[2];
-      }
-    }
+  }
 
-    // STEP 3: opening with resultat
-    function stepThree(id) {
-      var fy = getFiscalYear(id);
-      session.previous_fiscal_year = fy.fiscal_year_txt;
-      session.previous_fiscal_year_id = fy.id;
+  function checkClosing (close_fy, previous_fy_id) {
+    if (close_fy) {
+      stepThree(previous_fy_id);
+    } else {
+      $scope.createWithoutClosing = true;
+      $scope.step = steps[2];
+    }
+  }
 
-      getSolde(6, fy.id)
-      .then(function (data6) {
-        session.array6 = data6.data;
-        return getSolde(7, fy.id);
-      })
-      .then(function (data7) {
-        session.array7 = data7.data;
-      })
-      .then(function () {
-        session.solde6 = session.array6.reduce(sumDebMinusCred, 0);
-        session.solde7 = session.array7.reduce(sumCredMinusDeb, 0);
-        observation();
+  // STEP 3: opening with resultat
+  function stepThree(id) {
+    var fy = getFiscalYear(id);
+    session.previous_fiscal_year = fy.fiscal_year_txt;
+    session.previous_fiscal_year_id = fy.id;
+
+    getSolde(6, fy.id)
+    .then(function (data6) {
+      session.array6 = data6.data;
+      return getSolde(7, fy.id);
+    })
+    .then(function (data7) {
+      session.array7 = data7.data;
+      return getSolde(8, fy.id);
+    })
+    .then(function (data8) {
+      session.array8 = data8.data;
+      session.array8Charge = session.array8.filter(function (item) {
+        var prefix = Number(String(item.account_number).substr(0, 2));
+        return prefix % 2 !== 0 && item.is_charge === 1;
+      });
+      session.array8Profit = session.array8.filter(function (item) {
+        var prefix = Number(String(item.account_number).substr(0, 2));
+        return prefix % 2 === 0 && item.is_charge === 0;
       });
+    })
+    .then(function () {
+      session.solde6       = session.array6.reduce(sumDebMinusCred, 0);
+      session.solde7       = session.array7.reduce(sumCredMinusDeb, 0);
+      session.solde8Charge = session.array8Charge.reduce(sumDebMinusCred, 0);
+      session.solde8Profit = session.array8Profit.reduce(sumCredMinusDeb, 0);
+      session.charge       = session.solde6 + session.solde8Charge;
+      session.produit      = session.solde7 + session.solde8Profit;
+      observation();
+    });
```

Comment:
There is not error handling in this promise chain.  A simple `console.log(error)` is necessary and sufficient.